### PR TITLE
HOSTEDCP-1609: Support configuring Azure diagnostics via CLI flags

### PR DESCRIPTION
**Pull request by fxierh**
This is to facilitate automation which requires creating an HC/NP with Azure diagnostics enabled.

### CODE REVIEW

File: cmd/nodepool/azure/create.go [hunk 1]
```diff
@@ -56,6 +58,8 @@ func bindCoreOptions(opts *RawAzurePlatformCreateOptions, flags *pflag.FlagSet)
 	flags.StringVar(&opts.InstanceType, "instance-type", opts.InstanceType, "The instance type to use for the nodepool")
 	flags.Int32Var(&opts.DiskSize, "root-disk-size", opts.DiskSize, "The size of the root disk for machines in the NodePool (minimum 16)")
 	flags.StringVar(&opts.AvailabilityZone, "availability-zone", opts.AvailabilityZone, "The availabilityZone for the nodepool. Must be left unspecified if in a region that doesn't support AZs")
+	flags.StringVar(&opts.DiagnosticsStorageAccountType, "diagnostics-storage-account-type", opts.DiagnosticsStorageAccountType, "Specifies the type of storage account for storing diagnostics data. Supported values: Disabled, Managed, UserManaged.")
```
Comment (bryan-cox) [inline cmd/nodepool/azure/create.go:61]: We should add validation to enforce this

File: cmd/nodepool/azure/create.go [hunk 2]
```diff
@@ -56,6 +58,8 @@ func bindCoreOptions(opts *RawAzurePlatformCreateOptions, flags *pflag.FlagSet)
...
+	flags.StringVar(&opts.DiagnosticsStorageAccountURI, "diagnostics-storage-account-uri", opts.DiagnosticsStorageAccountURI, "Specifies the URI of the storage account for diagnostics data. Applicable only if --diagnostics-storage-account-type is set to UserManaged.")
```
Comment (bryan-cox) [inline cmd/nodepool/azure/create.go:62]: We should add validation to enforce this

File: cmd/nodepool/azure/create.go (hunk 1 above)
Comment (fxierh) [reply cmd/nodepool/azure/create.go]: Updated.

File: cmd/nodepool/azure/create.go (hunk 2 above)
Comment (fxierh) [reply cmd/nodepool/azure/create.go:63]: Updated.

**Comment by bryan-cox** [comment]
kubevirt will be blocked until https://github.com/openshift/hypershift/pull/4530

**Pull request merged**